### PR TITLE
xbmc/addons: cppcheck performance fixes

### PR DESCRIPTION
**Pull request by tobbi**
## Description

This fixes a few performance issues found with 
```
% cppcheck --enable=performance xbmc/addons --check-level=exhaustive --force
```

There's one more issue that I wasn't sure about (which seems a false positive):
```
xbmc/addons/addoninfo/AddonInfo.cpp:303:32: performance: Constructing a std::string_view from the result of c_str() is slow and redundant. [stlcstrConstructor]
        const std::string_view uid(filename.data() + startName.length());
```

## Motivation and context
This fixes issues reported by cppcheck.

## How has this been tested?
I compiled the code to see if I made mistakes while fixing cppcheck issues. The changes themselves should be pretty self-explanatory.

## What is the effect on users?
Eh, changes should be barely noticeable. 

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [X] All new and existing tests passed

### CODE REVIEW

**Comment by ksooo** [comment]
@tobbi please apply the formatting patches.

**Comment by ksooo** [comment]
jenkins build this please

File: xbmc/addons/LanguageResource.cpp [hunk 1]
```diff
@@ -29,12 +29,10 @@ namespace ADDON
 {
 
 CLanguageResource::CLanguageResource(const AddonInfoPtr& addonInfo)
-  : CResource(addonInfo, AddonType::RESOURCE_LANGUAGE)
+  : CResource(addonInfo, AddonType::RESOURCE_LANGUAGE),
+    // parse <extension> attributes
+    m_locale(CLocale::FromString(Type(AddonType::RESOURCE_LANGUAGE)->GetValue("@locale").asString()))
```
Comment (CastagnaIT) [inline xbmc/addons/LanguageResource.cpp:34]: was better readable where was before

File: xbmc/addons/ImageResource.cpp [hunk 2]
```diff
@@ -18,9 +18,9 @@ namespace ADDON
 {
 
 CImageResource::CImageResource(const AddonInfoPtr& addonInfo)
-  : CResource(addonInfo, AddonType::RESOURCE_IMAGES)
+  : CResource(addonInfo, AddonType::RESOURCE_IMAGES),
+    m_type(Type(AddonType::RESOURCE_IMAGES)->GetValue("@type").asString())
```
Comment (CastagnaIT) [inline xbmc/addons/ImageResource.cpp:22]: also this one as below, what kind of warning do the compiler for these two cases?

Comment (garbear) [reply xbmc/addons/ImageResource.cpp:22]: Yes, please include the warnings in the commit descriptions @tobbi

File: xbmc/addons/LanguageResource.cpp (hunk 1 above)
Comment (ksooo) [reply xbmc/addons/LanguageResource.cpp:34]: Like I said in the other PR, reason for the change is performance, not readability. (init members with desired value in init list)

File: xbmc/addons/ImageResource.cpp (hunk 2 above)
Comment (ksooo) [reply xbmc/addons/ImageResource.cpp:22]: Like I said in the other PR, reason for the change is performance, not readability. (init members with desired value in init list)

**Comment by tobbi** [comment]
jenkins build this please

**Comment by tobbi** [comment]
I've now rebased, added the cppcheck warnings in the commit description and applied the format patches.

**Review by garbear — APPROVED** [review]

**Pull request merged**